src/planner.cc

```diff
@@ -38,17 +47,81 @@ namespace {
 // The speed is initially the aimed goal; if it cannot be reached, the
 // AxisTarget will be modified to contain the actually reachable value. That is
 // used in planning along the path.
+// The starting_speed is the required speed at the start of the segment.
+// It is used to combine segments with different trajectory angles.
+// In the AxisTarget we define "motion invariants" (or targets). Values and
+// parameters that assuming the requested final trajectory stays the same, they
+// are constants, no matter the previous or next segments.
 struct AxisTarget {
   int position_steps[GCODE_NUM_AXES];  // Absolute position at end of segment.
-                                       // In steps.
+                                       // In steps. (steps)
 
   // Derived values
-  int delta_steps[GCODE_NUM_AXES];     // Difference to previous position.
+  int delta_steps[GCODE_NUM_AXES];  // Difference to previous position. (steps)
   enum GCodeParserAxis defining_axis;  // index into defining axis.
-  double speed;       // (desired) speed in steps/s on defining axis.
-  uint16_t aux_bits;  // Auxillary bits in this segment; set with M42
-  double dx, dy, dz;  // 3D delta_steps in real units
-  double len;         // 3D length
+  double start_speed;  // Starting speed of the segment (steps/s).
+  double speed;        // Maximum speed of the defining axis (steps/s).
+  double accel;        // Maximum acceleration of the defining axis (steps/s^2).
+  unsigned short aux_bits;  // Auxillary bits in this segment; set with M42
```

Comment:
This is where the clang-tidy complains: this shall be `uint16_t` to be more explicit about the width (as `short` is not really be defined).

Reply:
(solved)